### DISCUSSION TITLE
Available options to lspconfig

### DISCUSSION THREAD

**Discussion by breuerfelix** [VonHeikemen/lsp-zero.nvim#417] [Q&A] [answered]
Hey folks,
somehow i think that this might be the best place for me to find a good answer.
With neovim 0.10 there are some new cool features like inlay_hints.
I know that i can enable them via `vim.lsp.inlay_hint.enable()` dynamically. But how can i enable them per default in my lspconfig?

That brought me to the question: what are all the available options i have when configuring the LSP?
Usually i go to github, search for the plugin and see what the default options are. But this is not as easy with the lspconfig plugin.

Then i stumbled across [lazynvim](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/init.lua) and they set some really interesting options to the plugin.
Does anybody know where i see a list of full options to this plugin? They even type the object with `PluginLspOpts` but Google won't show me the definition of this class. I cannot find a single reference in neovim or lspconfig repository either.

**Comment by VonHeikemen** [comment, marked as answer]
>  what are all the available options i have when configuring the LSP?

It does have a type, here: [lspconfig.Config](https://github.com/neovim/nvim-lspconfig/blob/28b205ebe73a18f401e040585106f9bafd8ff21f/lua/lspconfig/configs.lua#L8-L16). It extends the built-in type [vim.lsp.ClientConfig](https://neovim.io/doc/user/lsp.html#vim.lsp.ClientConfig). The help page of lspconfig also provides some information: [lspconfig-setup](https://github.com/neovim/nvim-lspconfig/blob/28b205ebe73a18f401e040585106f9bafd8ff21f/doc/lspconfig.txt#L50).

**Comment by breuerfelix** [comment]
Thanks @VonHeikemen , but aren't these just the options you can pass to a language server .setup() call?
If you look at the link i posted, these are general LSP options on top level like `servers`, `inlay_hints`, `codelens`, ... I can't find any of these in the lspconfig.Config nor in the manual from the plugin. These are more like the options you pass to the plugin itself, and not to the individual language servers setup call.

/edit
These are custom options which are used in the `config()` function below in the document. So simple but i haven't figured that out after hours of research. Fresh mind now, some hours later, and this was obvious. 
Thanks for the hints to the correct configuration options :)
